spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkTruncateFunction.java

```diff
@@ -0,0 +1,377 @@
+    Assert.assertEquals(-10, scalarSql("SELECT system.truncate(10, %d)", -5));
+    Assert.assertEquals(-10, scalarSql("SELECT system.truncate(10, %d)", -10));
+    Assert.assertEquals(-20, scalarSql("SELECT system.truncate(10, %d)", -11));
+
+    // Check that different widths can be used
+    Assert.assertEquals(-2, scalarSql("SELECT system.truncate(2, %d)", -1));
+    Assert.assertEquals(0, scalarSql("SELECT system.truncate(300, %d)", 1));
+
+    Assert.assertEquals(
+        "Null input should return null",
+        null,
+        scalarSql("SELECT system.truncate(2, CAST(null AS int))"));
+  }
+
+  @Test
+  public void testTruncateBigInt() {
+    Assert.assertEquals(0L, scalarSql("SELECT system.truncate(10, 0L)"));
+    Assert.assertEquals(0L, scalarSql("SELECT system.truncate(10, 1L)"));
+    Assert.assertEquals(0L, scalarSql("SELECT system.truncate(10, 5L)"));
+    Assert.assertEquals(0L, scalarSql("SELECT system.truncate(10, 9L)"));
+    Assert.assertEquals(10L, scalarSql("SELECT system.truncate(10, 10L)"));
+    Assert.assertEquals(10L, scalarSql("SELECT system.truncate(10, 11L)"));
+    Assert.assertEquals(-10L, scalarSql("SELECT system.truncate(10, -1L)"));
+    Assert.assertEquals(-10L, scalarSql("SELECT system.truncate(10, -5L)"));
+    Assert.assertEquals(-10L, scalarSql("SELECT system.truncate(10, -10L)"));
+    Assert.assertEquals(-20L, scalarSql("SELECT system.truncate(10, -11L)"));
+
+    // Check that different widths can be used
+    Assert.assertEquals(-2L, scalarSql("SELECT system.truncate(2, -1L)"));
+
+    Assert.assertEquals(
+        "Null input should return null",
+        null,
+        scalarSql("SELECT system.truncate(2, CAST(null AS bigint))"));
+  }
+
+  @Test
+  public void testTruncateDecimal() {
+    // decimal truncation works by applying the decimal scale to the width: ie 10 scale 2 = 0.10
+    Assert.assertEquals(
+        new BigDecimal("12.30"),
+        scalarSql("SELECT system.truncate(10, CAST(%f as DECIMAL(9, 2)))", 12.34));
+
+    Assert.assertEquals(
+        new BigDecimal("12.30"),
+        scalarSql("SELECT system.truncate(10, CAST(%f as DECIMAL(9, 2)))", 12.30));
+
+    Assert.assertEquals(
+        new BigDecimal("12.290"),
+        scalarSql("SELECT system.truncate(10, CAST(%f as DECIMAL(9, 3)))", 12.299));
+
+    Assert.assertEquals(
+        new BigDecimal("0.03"),
+        scalarSql("SELECT system.truncate(3, CAST(%f as DECIMAL(5, 2)))", 0.05));
+
+    Assert.assertEquals(
+        new BigDecimal("0.00"),
+        scalarSql("SELECT system.truncate(10, CAST(%f as DECIMAL(9, 2)))", 0.05));
+
+    Assert.assertEquals(
+        new BigDecimal("-0.10"),
+        scalarSql("SELECT system.truncate(10, CAST(%f as DECIMAL(9, 2)))", -0.05));
+
+    Assert.assertEquals(
+        "Implicit decimal scale and precision should be allowed",
+        new BigDecimal("12345.3480"),
+        scalarSql("SELECT system.truncate(10, 12345.3482)"));
+
+    int scale = 4;
```

Comment:
Why have a variable for this if it isn't used everywhere?

Reply:
Removed.